_posts/2024-10-15-anaconda-defaults.md

```diff
@@ -0,0 +1,175 @@
+
+The safest option is to install the `conda-forge` distribution. This has the added benefit of installing `mamba` from the very beginning and is the [recommended method of installation](https://mamba.readthedocs.io/en/latest/installation/mamba-installation.html) by the `mamba` devs.
+
+To test your setup, remove the `defaults` channel (see above) try to install `fonts-anaconda`, which is accessible only via `defaults`. Warning: this will access the Anaconda defaults channel! Also check to see if you are using the `classic` or `libmamba` solver by running `conda config --show | grep "solver:"`
+
+It is far simpler to install `miniforge3`.
+
+### Protecting against `defaults` channel
+
+It's possible to protect users of your pipelines/tools by including `nodefaults` channel in a `conda.yaml` file. For example:
+
+```
+channels:
+  - conda-forge
+  - bioconda
+  - nodefaults
+```
+
+This will [override the defaults channel](https://docs.conda.io/projects/conda/en/4.6.1/user-guide/tasks/manage-environments.html#creating-an-environment-file-manually) if it exists in the users global config. Unfortunately, [this is specific to `conda env` subcommand](https://stackoverflow.com/a/67708768), therefore it will not work for `conda install` or `conda create` when added to global config. There is an open [issue](https://github.com/conda/conda/issues/12010) on github for this feature.
+
+## FAQs + FYIs
+
+### - How can I transition safely from `defaults` channel?
+
+If you are worried removing `defaults` will break your current setup, `conda-forge` [has great documentation](https://conda-forge.org/docs/user/transitioning_from_defaults/) on how to test and transition from your addiction to `defaults`.
+
+### - Be wary of foreign `conda.yaml` files in software/pipelines
+
+Lots of pipelines/tools will use `conda.yaml` to enable easy installation of dependencies. It's very likely `defaults` could be in a `conda.yaml` file because it's been the... default to include it.
+
+Unfortunately as far as I know, there is no setting or method available to protect against a `conda.yaml` using a package from `defaults`.
+
+There is a feature request for `nodefaults` [to apply everywhere](https://github.com/conda/conda/issues/12010). 
+
+I suggest always double checking foreign `conda.yaml` files before installing.
+
+### - Will removing `defaults` interfere with the install of bioinfo tools?
+
+Very unlikely. `conda-forge` [transitioned away](https://conda-forge.org/news/2021/09/30/defaults-channel-is-now-dropped-when-building-conda-forge-packages/) from Anaconda's `defaults` channel in 2021 and has continued to diverge in both names and recipes. There is a slight chance it may cause issues for old pre 2020/2021 recipes but this is rare as recipies have continually been updated. Not too worry though, `conda-forge` is community driven, [feel free to contribute if you run into any problems!](https://conda-forge.org/docs/user/contributing/)
```

Comment:
```suggestion
Very unlikely. `conda-forge` [transitioned away](https://conda-forge.org/news/2021/09/30/defaults-channel-is-now-dropped-when-building-conda-forge-packages/) from Anaconda's `defaults` channel in 2021 and has continued to diverge in both names and recipes. There is a slight chance it may cause issues for old pre 2020/2021 recipes but this is rare as recipes have continually been updated. Not too worry though, `conda-forge` is community driven, [feel free to contribute if you run into any problems!](https://conda-forge.org/docs/user/contributing/)
```